module1/homework/make_unique/makeUnique-ut.cpp

```diff
@@ -0,0 +1,90 @@
+
+#include <gtest/gtest.h>
+#include <tuple>
+#include "makeUnique.hpp"
+
+namespace TestConsts {
+constexpr int intValue = 42;
+constexpr double doubleValue = 4.2;
+constexpr bool boolValue = false;
+constexpr int lValueInt = -1;
+constexpr int rValueInt = 1;
+}  // namespace TestConsts
+
+struct TestData {
+    TestData(int f, double s, bool t)
+        : first_(f), second_(s), third_(t) {}
+    int first_ = 4;
+    double second_ = 4.2;
+    bool third_ = false;
+};
+
+struct LRValue {
+    LRValue(int value)
+        : value_(value){};
+    LRValue(LRValue&)
+        : value_(TestConsts::lValueInt){};
+    LRValue(LRValue&&)
+        : value_(TestConsts::rValueInt){};
+
+    int value_;
+};
+
+TEST(makeUnique, shouldCreatePointerToInt) {
+    auto ptr = myMakeUnique::make_unique<int>(TestConsts::intValue);
+
+    ASSERT_EQ(*ptr, TestConsts::intValue);
+}
+
+TEST(makeUnique, shouldCreatePointerToDouble) {
+    auto ptr = myMakeUnique::make_unique<double>(TestConsts::doubleValue);
+
+    ASSERT_EQ(*ptr, TestConsts::doubleValue);
+}
+
+TEST(makeUnique, shouldCreatePointerToIntWithDefaultValue) {
+    auto ptr = myMakeUnique::make_unique<int>();
+
+    ASSERT_EQ(*ptr, int());
+}
+
+TEST(makeUnique, shouldCreatePointerToTuple) {
+    auto ptr = myMakeUnique::make_unique<std::tuple<int, double>>(
+        std::make_tuple(TestConsts::intValue, TestConsts::doubleValue));
```

Comment:
std::tuple with std::make_tuple is redundant. std::make_tuple returns just std::tuple. Get rid of one of them.